packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -355,18 +355,19 @@ IOSink? _openSink(String? output) {
   return sink;
 }
 
-/// A generator that will write code to a sink based on the contents of [PigeonOptions].
-abstract class Generator {
-  /// Returns an [IOSink] instance to be written to if the [Generator] should
-  /// generate.  If it returns `null`, the [Generator] will be skipped.
+/// An adapter that will call a generator to write code to a sink
+/// based on the contents of [PigeonOptions].
+abstract class Adapter {
```

Comment:
If you make Adapter a concrete class and Generator stateless like I mentioned in the other PR you'll find that there is a lot of boilerplate and duplication you could eliminate.

Eventually the list of adapters could look like this:
```dart
adapters = <Adapter>[
  Adapter(const CppHeaderGenerator()),
  Adapter(const CppSourceGenerator()),
  Adapter(const DartGenerator()),
];
```